components/ws-manager/pkg/manager/monitor.go

```diff
@@ -512,25 +509,14 @@ func (m *Monitor) writeEventTraceLog(status *api.WorkspaceStatus, wso *workspace
 	json.NewEncoder(out).Encode(entry)
 }
 
-// traceWorkspace updates the workspace span if the workspace has OpenTracing information associated with it.
-// The resulting context may be associated with trace information that can be used to trace the effects of this status
-// update throughout the rest of the system.
-func (m *Monitor) traceWorkspace(occasion string, wso *workspaceObjects) opentracing.Span {
-	var traceID string
-	if traceID == "" && wso.Pod != nil {
-		traceID = wso.Pod.Annotations[wsk8s.TraceIDAnnotation]
-	}
-	spanCtx := tracing.FromTraceID(traceID)
-	if spanCtx == nil {
-		// no trace information available
-		return opentracing.NoopTracer{}.StartSpan("noop")
-	}
-
-	span := opentracing.StartSpan(fmt.Sprintf("/workspace/%s", occasion), opentracing.FollowsFrom(spanCtx))
+// traceWorkspaceState creates a new span that records the phase of workspace
+func (m *Monitor) traceWorkspaceState(state string, wso *workspaceObjects) opentracing.Span {
+	span := opentracing.StartSpan(fmt.Sprintf("/workspace/%s", state))
```

Comment:
This span is no longer connected to the workspace. We should add the instance/workspace ID as label so that we can correlate.

Reply:
Shouldn't this add it?
https://github.com/gitpod-io/gitpod/blob/f636456dcfd32a13b09b0f3b8e4f177f646a1dc6/components/ws-manager/pkg/manager/monitor.go#L515-L517

Reply:
it does - my bad